src/main/java/com/manufacturing/dao/jdbc/CarDaoJdbcImpl.java

```diff
@@ -0,0 +1,203 @@
+package com.manufacturing.dao.jdbc;
+
+import com.manufacturing.dao.CarDao;
+import com.manufacturing.lib.Dao;
+import com.manufacturing.model.Car;
+import com.manufacturing.model.Driver;
+import com.manufacturing.model.Manufacturer;
+import com.manufacturing.util.ConnectionUtil;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class CarDaoJdbcImpl implements CarDao {
+    @Override
+    public Car create(Car car) {
+        String query = "INSERT INTO cars (manufacturer_id, model) VALUE (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            statement.setLong(1, car.getManufacturer().getId());
+            statement.setString(2, car.getModel());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                car.setId(resultSet.getObject(1, Long.class));
+            }
+            return car;
+        } catch (SQLException e) {
+
+            throw new DataProcessingException("This data can't be added to table\n"
+                    + e.getMessage(), e);
+        }
+    }
+
+    @Override
+    public Optional<Car> get(Long carId) {
+        String query = "SELECT c.car_id, model, c.manufacturer_id,"
+                + " manufacturer_name, origin\n"
+                + "FROM cars c\n"
+                + "INNER JOIN manufacturers m "
+                + "ON m.manufacturer_id = c.manufacturer_id\n"
+                + "WHERE c.car_id = ? AND c.deleted = false;";
+        Car car = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, carId);
+            ResultSet resultSet = statement.executeQuery();
+            List<Driver> drivers = getDrivers(carId, connection);
+            if (resultSet.next()) {
+                car = getCar(resultSet, connection);
+                car.setDrivers(drivers);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("We can't get car with ID:" + carId, e);
+        }
+        return Optional.ofNullable(car);
+    }
+
+    @Override
+    public Car update(Car car) {
+        String insertQuery = "INSERT INTO cars_drivers (driver_id, car_id) VALUES (?, ?)";
+        String query = "UPDATE cars SET manufacturer_id = ?, model = ? "
+                + "WHERE car_id = ? AND deleted = false";
```

Comment:
What if some drivers were removed from this car?
Let's make this update in three steps:
1) Update car by itself;
2) remove old drivers for this car from cars_drivers;
3) insert the new list;

Reply:
I have method `deleteDriversByID()` for this

Reply:
Remove method DeleteDriversByID() and move it functional to method update